transports/janus_websockets.c

```diff
@@ -790,17 +794,46 @@ static int janus_websockets_callback(struct libwebsocket_context *this,
 			}
 			if(!ws_client->destroy && !g_atomic_int_get(&stopping)) {
 				janus_mutex_lock(&ws_client->mutex);
-				/* Shoot all the pending messages first */
+				/* Check if we have a pending/partial write to complete first */
+				if(ws_client->partial_message && ws_client->partial_offset > 0
+						&& !ws_client->destroy && !g_atomic_int_get(&stopping)) {
+					int pending = strlen(ws_client->partial_message) - ws_client->partial_offset;
+					JANUS_LOG(LOG_VERB, "[WSS-%p] Completing pending WebSocket write (still need to write last %d/%zu bytes)...\n",
+						wsi, pending, strlen(ws_client->partial_message));
+					int sent = libwebsocket_write(wsi, (unsigned char *)(ws_client->partial_message + ws_client->partial_offset), pending, LWS_WRITE_TEXT);
```

Comment:
The partial_message doesn't include the post-padding.
The easy fix is to set `partial_message = buf` instead of `partial_message = response`, below.

But I guess that if less than the response was sent, the 4 bytes in buf after the last byte that was sent are clobbered by the libwebsocket framing, and response needs to be copied to buf again.

I also wonder if it wouldn't be reasonable to pick a maximum size for buf, and make it an array on the stack, to avoid the extra malloc/free.